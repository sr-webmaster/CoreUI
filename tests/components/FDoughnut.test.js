import { mount, createLocalVue } from '@vue/test-utils'
import Component from '../../src/components/FDoughnut'

const dataSet = [10, 20, 30]
const labels = ['Red', 'Green', 'Blue']
const colors = ['red', 'green', 'blue']

describe('Doughnut Component', () => {
  let localVue

  beforeEach(() => {
    localVue = createLocalVue()
  })

  describe('Visual', () => {
    test('chartData set', () => {
      const wrapper = mount(Component, {
        localVue,
        propsData: {
          chartData: {
            datasets: [{
              data: dataSet,
              backgroundColor: colors,
              borderWidth: 0
            }],
            labels: labels
          },
          options: {
            legend: {
              display: false
            },
            cutoutPercentage: 70
          }
        }
      })
      expect(wrapper.element).toMatchSnapshot()
    })
    test('chartData and some options set', () => {
      const wrapper = mount(Component, {
        localVue,
        propsData: {
          chartData: {
            datasets: [{
              data: dataSet,
              backgroundColor: colors,
              borderWidth: 5
            }],

            labels: labels
          },
          options: {
            legend: {
              display: false
            },
            cutoutPercentage: 70
          }
        }
      })
      expect(wrapper.element).toMatchSnapshot()
    })
    test('chartData set, withLabels', () => {
      const wrapper = mount(Component, {
        localVue,
        propsData: {
          chartData: {
            datasets: [{
              data: dataSet,
              backgroundColor: colors,
              borderWidth: 0
            }],

            labels: labels
          },
          options: {
            legend: {
              display: true
            },
            cutoutPercentage: 70
          }
        }
      })
      expect(wrapper.element).toMatchSnapshot()
    })
    test('chartData set, withLabels and withPercentage', () => {
      const wrapper = mount(Component, {
        localVue,
        propsData: {
          chartData: {
            datasets: [{
              data: dataSet,
              backgroundColor: colors,
              borderWidth: 0
            }],

            labels: labels,
            withPercentage: true
          },
          options: {
            legend: {
              display: true
            },
            cutoutPercentage: 70
          }
        }
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })
  describe('Method', () => {
    test('getPercentageLabels()', () => {
      const wrapper = mount(Component, {
        localVue,
        propsData: {
          chartData: {
            datasets: [{
              data: dataSet,
              backgroundColor: colors,
              borderWidth: 0
            }],
            labels: labels
          },
          options: {
            legend: {
              display: false
            },
            cutoutPercentage: 70
          }
        }
      })
      expect(wrapper.vm.getPercentageLabels(labels, dataSet)).toEqual(['Red 16.7%', 'Green 33.3%', 'Blue 50%'])
    })
  })
})
