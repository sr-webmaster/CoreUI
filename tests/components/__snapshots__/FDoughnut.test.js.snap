// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Doughnut Component Visual chartData and some options set 1`] = `
<div
  class=""
>
  <canvas
    class="chartjs-render-monitor"
    height="0"
    id="doughnut-chart"
    style="display: block;"
    width="0"
  />
</div>
`;

exports[`Doughnut Component Visual chartData set 1`] = `
<div
  class=""
>
  <canvas
    class="chartjs-render-monitor"
    height="0"
    id="doughnut-chart"
    style="display: block;"
    width="0"
  />
</div>
`;

exports[`Doughnut Component Visual chartData set, withLabels 1`] = `
<div
  class=""
>
  <canvas
    class="chartjs-render-monitor"
    height="0"
    id="doughnut-chart"
    style="display: block;"
    width="0"
  />
</div>
`;

exports[`Doughnut Component Visual chartData set, withLabels and withPercentage 1`] = `
<div
  class=""
>
  <canvas
    class="chartjs-render-monitor"
    height="0"
    id="doughnut-chart"
    style="display: block;"
    width="0"
  />
</div>
`;
